src/server/pfs/server/driver.go

```diff
@@ -765,7 +765,9 @@ func (d *driver) writeFinishedCommit(ctx context.Context, commit *pfs.Commit, co
 				if err := d.branches(commit.Repo.Name).ReadWrite(stm).Get(branch.Name, branchInfo); err != nil {
 					return err
 				}
-				if branchInfo.Head.ID == commit.ID {
+				// If the head commit of master has been deleted, we could get here if another branch
+				// had shared its head commit with master, and then we created a new commit on that branch
+				if branchInfo.Head != nil && branchInfo.Head.ID == commit.ID {
```

Comment:
Actually, I think there is a cleaner solution to this. We can just check if the commit's branch is the master branch -- since we keep track of that now.

Reply:
So we keep track if a commit was created on master, but I think it's possible in weird cases for the head of master not to have been created on master. For example in the deferred processing case all commits get created on `staging`

Reply:
Oh right, that's true